internal/apis/certmanager/types_issuer.go

```diff
@@ -232,6 +238,26 @@ type VaultAppRole struct {
 	SecretRef cmmeta.SecretKeySelector
 }
 
+// VaultKubernetesAuth is used to authenticate against Vault using a client
+// certificate stored in a Secret.
+type VaultClientCertificateAuth struct {
+	// The Vault mountPath here is the mount path to use when authenticating with
+	// Vault. For example, setting a value to `/v1/auth/foo`, will use the path
+	// `/v1/auth/foo/login` to authenticate with Vault. If unspecified, the
+	// default value "/v1/auth/cert" will be used.
+	// +optional
+	Path string
+
+	// Reference to Kubernetes Secret of type "kubernetes.io/tls" (hence containing
+	// tls.crt and tls.key) used to authenticate to Vault using TLS client
+	// authentication.
+	// +optional
```

Comment:
Should make this non-optional.

Reply:
Actually, keeping this optional to allow using a client certificate that is supplied using an (as of yet AFAIK non-existing) implementation of #4774.

Explained more thoroughly:

#4774 would primarily target security at "transport layer" while also using another authentication method.

But, setting a client certificate via #4774 **and** using the client certificate authentication method with empty `secretName` would allow using the client certificate for all requests (at "transport layer") as well as for the authentication.

Reply:
As already written in https://github.com/cert-manager/cert-manager/pull/4330#issuecomment-1020403227 ...

Reply:
That makes sense.